packages/manager/src/features/CloudPulse/shared/CloudPulseRegionSelect.test.tsx

```diff
@@ -1,24 +1,48 @@
+import userEvent from '@testing-library/user-event';
 import * as React from 'react';
 
-import * as regions from 'src/queries/regions/regions';
+import { dashboardFactory, regionFactory } from 'src/factories';
 import { renderWithTheme } from 'src/utilities/testHelpers';
 
 import { CloudPulseRegionSelect } from './CloudPulseRegionSelect';
 
 import type { CloudPulseRegionSelectProps } from './CloudPulseRegionSelect';
 import type { Region } from '@linode/api-v4';
+import type { CloudPulseServiceTypeMapFlag, Flags } from 'src/featureFlags';
+import type * as regions from 'src/queries/regions/regions';
 
 const props: CloudPulseRegionSelectProps = {
   handleRegionChange: vi.fn(),
   label: 'Region',
   selectedDashboard: undefined,
 };
 
-describe('CloudPulseRegionSelect', () => {
-  vi.spyOn(regions, 'useRegionsQuery').mockReturnValue({
-    data: Array<Region>(),
-  } as ReturnType<typeof regions.useRegionsQuery>);
+const queryMocks = vi.hoisted(() => ({
+  useRegionsQuery: vi.fn().mockReturnValue({}),
+}));
+
+const flags: Partial<Flags> = {
+  aclpServiceTypeMap: [
+    {
+      serviceType: 'dbaas',
+      supportedRegionIds: 'us-west',
```

Comment:
May be can we add more regions in array , so that testing with multiple regions would be validated.